### ISSUE TITLE
Unable to create 2GP managed package

### ISSUE THREAD

**Issue by aozomaro** [forcedotcom/cli#2494]
> **Note** 
> Before you submit your issue, make sure that:
> - You're using the latest version of Salesforce CLI.
> - You've searched both open and closed issues for related posts.
> - You've used the `doctor` command to diagnose common issues. 
> - You understand that GitHub Issues don't adhere to any agreement or SLA.
>   - If you require immediate assistance, use official channels such as Salesforce Customer Support.

### Summary
When we try to create a managed package we get an error which indicates that the package.xml API version is invalid. More specifically we receive this error message:
` Multiple errors occurred: \n(1) package.xml: Invalid version specified:59.0`

### Steps To Reproduce

1. Generate a managed package release by running the following command: ```sfdx package:version:create -p "Our Package" -v dev-hub -c -w 60 --verbose --json```

### Expected result
Build should create package successfully

### Actual result
` Multiple errors occurred: \n(1) package.xml: Invalid version specified:59.0`

### System Information
```json
{
"cliVersion": "@salesforce/cli/2.8.11",
"architecture": "linux-x64",
"nodeVersion": "node-v18.14.0",
"osVersion": "Linux 5.15.0-1028-aws",
"shell": "bash",
"rootPath": "/opt/circleci/.nvm/versions/node/v18.14.0/lib/node_modules/@salesforce/cli",
"pluginVersions": [
"@oclif/plugin-autocomplete 2.3.8 (core)",
"@oclif/plugin-commands 2.2.25 (core)",
"@oclif/plugin-help 5.2.19 (core)",
"@oclif/plugin-not-found 2.4.1 (core)",
"@oclif/plugin-plugins 3.4.2 (core)",
"@oclif/plugin-search 0.0.22 (core)",
"@oclif/plugin-update 3.2.3 (core)",
"@oclif/plugin-version 1.3.10 (core)",
"@oclif/plugin-warn-if-update-available 2.1.0 (core)",
"@oclif/plugin-which 2.2.32 (core)",
"@salesforce/cli 2.8.11 (core)",
"apex 2.3.14 (core)",
"auth 2.8.16 (core)",
"data 2.5.8 (core)",
"deploy-retrieve 1.17.8 (core)",
"info 2.6.40 (core)",
"limits 2.3.33 (core)",
"login 1.2.29 (core)",
"marketplace 0.1.3 (core)",
"org 2.10.6 (core)",
"schema 2.3.25 (core)",
"settings 1.4.28 (core)",
"sobject 0.2.6 (core)",
"source 2.10.33 (core)",
"telemetry 2.3.1 (core)",
"templates 55.5.11 (core)",
"trust 2.6.9 (core)",
"user 2.3.32 (core)"
]
}
```

**Comment by aozomaro** [comment]
Thanks. I updated the CLI and received the same error. Here is the output. 

{ "cliVersion": "@salesforce/cli/2.10.2", "architecture": "linux-x64", "nodeVersion": "node-v18.14.0", "osVersion": "Linux 5.15.0-1028-aws", "shell": "bash", "rootPath": "/opt/circleci/.nvm/versions/node/v18.14.0/lib/node_modules/@salesforce/cli", "pluginVersions": [ "@oclif/plugin-autocomplete 2.3.8 (core)", "@oclif/plugin-commands 2.2.25 (core)", "@oclif/plugin-help 5.2.19 (core)", "@oclif/plugin-not-found 2.4.1 (core)", "@oclif/plugin-plugins 3.7.1 (core)", "@oclif/plugin-search 0.0.22 (core)", "@oclif/plugin-update 3.2.3 (core)", "@oclif/plugin-version 1.3.10 (core)", "@oclif/plugin-warn-if-update-available 2.1.0 (core)", "@oclif/plugin-which 2.2.32 (core)", "@salesforce/cli 2.10.2 (core)", "apex 2.3.15 (core)", "auth 2.8.17 (core)", "data 2.5.9 (core)", "deploy-retrieve 1.18.0 (core)", "info 2.6.43 (core)", "limits 2.3.34 (core)", "login 1.2.31 (core)", "marketplace 0.2.2 (core)", "org 2.10.8 (core)", "schema 2.3.26 (core)", "settings 1.4.29 (core)", "sobject 0.2.7 (core)", "source 2.10.34 (core)", "telemetry 2.3.3 (core)", "templates 55.5.12 (core)", "trust 2.6.13 (core)", "user 2.3.32 (core)" ] }

**Comment by shetzel** [comment]
Has your devhub been updated to API version 59.0?  What does your sfdx-project.json look like (redacting anything you prefer to be private), specifically with any API versions or sourceApiVersion?

**Comment by aozomaro** [comment]
> Has your devhub been updated to API version 59.0? What does your sfdx-project.json look like (redacting anything you prefer to be private), specifically with any API versions or sourceApiVersion?

Thanks for getting back to me.

* Dev hub is on the NA156 instance, which is reported to be on Summer '23 release: https://status.salesforce.com/instances/NA156

* sfdx-project.json:

```json
{
  "packageDirectories": [
    {
      "path": "force-app",
      "default": true,
      "package": "My Package",
      "ancestorVersion": "2.82.0",
      "versionNumber": "2.83.0.NEXT",
      "postInstallScript": "PostInstall",
      "branch": "master",
      "unpackagedMetadata": {
        "path": "unpackaged"
      },
      "apexTestAccess": {
        "permissionSets": [ 
              "PermSet1"
        ]
      }
    },
    {
      "path": "unpackaged",
      "default": false
    }
  ],
  "name": "someAppName",
  "namespace": "SomeNamespace",
  "sourceApiVersion": "58.0",
  "packageAliases": {
    "My Package": "somePackageId",
    "My Package@2.83.0-1-master": "someVersionId"
  }
}

```

**Comment by db-lyon** [comment]
I also can't create a 2GP managed package. Not sure if it's related..

`sf package version create...` successfully runs, but when I check the status...
sf package:version:create:report -i 08cJx0000REDACTED --json > result.json    

I get:

````{
  "code": 1,
  "context": "PackageVersionCreateReportCommand",
  "commandName": "PackageVersionCreateReportCommand",
  "message": "Cannot read properties of undefined (reading 'Status')",
  "name": "TypeError",
  "status": 1,
  "stack": "TypeError: Cannot read properties of undefined (reading 'Status')\n    at PackageVersionCreateReportCommand.display (C:\\Users\\david\\AppData\\Local\\sf\\node_modules\\@salesforce\\plugin-packaging\\lib\\commands\\package\\version\\create\\report.js:31:40)\n    at PackageVersionCreateReportCommand.run (C:\\Users\\david\\AppData\\Local\\sf\\node_modules\\@salesforce\\plugin-packaging\\lib\\commands\\package\\version\\create\\report.js:27:14)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async PackageVersionCreateReportCommand._run (C:\\Users\\david\\AppData\\Local\\sf\\node_modules\\@oclif\\core\\lib\\command.js:117:22)\n    at async Config.runCommand (C:\\Users\\david\\AppData\\Local\\sf\\client\\2.10.2-1c35561\\node_modules\\@oclif\\core\\lib\\config\\config.js:314:25)\n    at async run (C:\\Users\\david\\AppData\\Local\\sf\\client\\2.10.2-1c35561\\node_modules\\@oclif\\core\\lib\\main.js:89:16)",
  "exitCode": 1,
  "warnings": []
}

**Comment by aozomaro** [comment]
My issue has been resolved. 

Th following blurb from SFDC support i what fixed the issue. 

> Please try to use "released: previous" and apiVersio 58.0. then you can enforce this package created in apiversion 58.0 and that released version.
